Src/EventCounterCollector/EventCounterCollector/EventCounterListener.cs

```diff
@@ -0,0 +1,230 @@
+                {
+                    IDictionary<string, object> eventPayload = eventData.Payload[0] as IDictionary<string, object>;
+                    if (eventPayload != null)
+                    {
+                        this.ExtractAndPostMetric(eventData.EventSource.Name, eventPayload);
+                    }
+                    else
+                    {
+                        EventCounterCollectorEventSource.Log.IgnoreEventWrittenAsEventPayloadNotParseable(eventData.EventSource.Name);
+                    }
+                }
+                else
+                {
+                    EventCounterCollectorEventSource.Log.IgnoreEventWrittenAsEventSourceNotInConfiguredList(eventData.EventSource.Name);
+                }
+            }
+            else
+            {
+                EventCounterCollectorEventSource.Log.IgnoreEventWrittenAsNotInitialized(eventData.EventSource.Name);
+            }
+        }
+
+        private void EnableIfRequired(EventSource eventSource)
+        {
+            try
+            {
+                // The EventSourceName is in the list we want to collect some counters from.
+                if (this.countersToCollect.ContainsKey(eventSource.Name))
+                {                    
+                    // Unlike regular Events, the only relevant parameter here for EventCounter is the dictionary containing EventCounterIntervalSec.
+                    this.EnableEvents(eventSource, this.level, (EventKeywords)(-1), refreshIntervalDictionary);
+
+                    EventCounterCollectorEventSource.Log.EnabledEventSource(eventSource.Name);
+                }
+                else
+                {
+                    EventCounterCollectorEventSource.Log.NotEnabledEventSource(eventSource.Name);
+                }
+            }
+            catch (Exception ex)
+            {
+                EventCounterCollectorEventSource.Log.EventCounterCollectorError("EventCounterListener EnableEventSource", ex.Message);
+            }
+        }
+
+        private void ExtractAndPostMetric(string eventSourceName, IDictionary<string, object> eventPayload)
+        {
+            try
+            {
+                MetricTelemetry metricTelemetry = new MetricTelemetry();
+                bool calculateRate = false;
+                double actualValue = 0.0;
+                double actualInterval = 0.0;
+                int actualCount = 0;
+                foreach (KeyValuePair<string, object> payload in eventPayload)
+                {
+                    var key = payload.Key;
+                    if (key.Equals("Name", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var counterName = payload.Value.ToString();
+                        if (this.countersToCollect[eventSourceName].Contains(counterName))
+                        {
+                            metricTelemetry.Name = eventSourceName + "|" + counterName;
```

Comment:
Is this what appears to the users? If so you should use the `DisplayName` property in the payload instead. Our guidance is that `Name` is the property that should be used internally within code, command-line parameters, etc. whereas `DisplayName` property is the one that gets shown on APM/dashboards/etc.

Reply:
Yes! good point. I somehow missed this. Will do a followup PR to address this and actually allow users to specific 'reportAs' as well, so users can decide what the name should appear.